docs/guides/01_creating_an_aggregate.md

```diff
@@ -0,0 +1,190 @@
+# Creating an aggregate
+
+## Intro
+
+Aggregates, as defined by Domain Driven Design, are clusters of related objects which can be treated as a single domain entity. One of these objects is distinguished as an aggregate root. Every object of an aggregate must be accessed through the aggregate root. This guide will present how to create a simple aggregate using LeanCode Core Library.
+
+## Scenario
+
+Let's imagine a simple project management app where users can create projects which contain tasks to which people can be assigned. We will create two aggregates - one representing a project and the other representing a person which can be assigned to the task.
+
+## Aggregates
+
+Let's start with a simple model for the project aggregate.
+
+```csharp
+public class Project : IAggregateRoot<SId<Project>>
+{
+    private readonly List<Task> tasks = new();
+
+    public SId<Project> Id { get; private init; }
+    public string Name { get; private set; }
+
+    public IReadOnlyList<Task> Tasks => tasks;
+
+    DateTime IOptimisticConcurrency.DateModified { get; set; }
+
+    private Project() { }
+
+    public static Project Create(string name)
+    {
+        return new Project
+        {
+            Id = SId<Project>.New(),
+            Name = name,
+        };
+    }
+}
+```
+
+As you can see, the class implements `IAggregateRoot` interface - it marks the class as being the root of an aggregate. Moreover the `Id` field of the class is of type `SId<Project>` - it is a special type present in the core library which is basically a Guid prefixed with a class name. In this case, the Id of the Project will look somewhat like `project_45a8f39f-9df0-4a23-b781-2a46de22fac1`.
+The `Project` also has a list of `Tasks`. Notice that there are two lists containing tasks of a project - the `Tasks` one is a readonly interface for the `tasks` which contents can be changed by the class.
```

Comment:
I'd stress that we design the API in a way that we cannot manipulate the aggregate from the outside world (only object can modify it's state based on the method it exposes)